text/0000-otlp-trace-data-format.md

```diff
@@ -0,0 +1,376 @@
+    SPAN_KIND_UNSPECIFIED = 0;
+
+    // Indicates that the span is used internally. Default value.
+    INTERNAL = 1;
+
+    // Indicates that the span covers server-side handling of an RPC or other
+    // remote network request.
+    SERVER = 2;
+
+    // Indicates that the span covers the client-side wrapper around an RPC or
+    // other remote request.
+    CLIENT = 3;
+
+    // Indicates that the span describes producer sending a message to a broker.
+    // Unlike client and  server, there is no direct critical path latency relationship
+    // between producer and consumer spans.
+    PRODUCER = 4;
+
+    // Indicates that the span describes consumer recieving a message from a broker.
+    // Unlike client and  server, there is no direct critical path latency relationship
+    // between producer and consumer spans.
+    CONSUMER = 5;
+  }
+
+  // Distinguishes between spans generated in a particular context. For example,
+  // two spans with the same name may be distinguished using `CLIENT` (caller)
+  // and `SERVER` (callee) to identify queueing latency associated with the span.
+  SpanKind kind = 7;
+
+  // The start time of the span. On the client side, this is the time kept by
+  // the local machine where the span execution starts. On the server side, this
+  // is the time when the server's application handler starts running.
+  //
+  // This field is semantically required. When not set on receive -
+  // receiver should set it to the value of end_time field if it was
+  // set. Or to the current time if neither was set. It is important to
+  // keep end_time > start_time for consistency.
+  //
+  // This field is required.
+  int64 start_time_unixnano = 8;
+
+  // The end time of the span. On the client side, this is the time kept by
+  // the local machine where the span execution ends. On the server side, this
+  // is the time when the server application handler stops running.
+  //
+  // This field is semantically required. When not set on receive -
+  // receiver should set it to start_time value. It is important to
+  // keep end_time > start_time for consistency.
+  //
+  // This field is required.
+  int64 end_time_unixnano = 9;
+
+  // The set of attributes. The value can be a string, an integer, a double
+  // or the Boolean values `true` or `false`. Note, global attributes like
+  // server name can be set as tags using resource API. Examples of attributes:
+  //
+  //     "/http/user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"
+  //     "/http/server_latency": 300
+  //     "abc.com/myattribute": true
+  //     "abc.com/score": 10.239
+  repeated AttributeKeyValue attributes = 10;
+
+  // A time-stamped event in the Span.
+  message TimedEvent {
+    // The time the event occurred.
+    int64 time_unixnano = 1;
+
+    // A text annotation with a set of attributes.
+    message Event {
```

Comment:
Why do you need an extra message for this? By removing this we get the same performance as keeping the Timestamp :)

Reply:
Good point. Removing. I'll also post benchmarks for spans with TimedEvents (current benchmark doesn't have any).

Reply:
Changing to:
```
  message TimedEvent {
    // The time the event occurred.
    int64 time_unixnano = 1;

    // A user-supplied name describing the event.
    string name = 2;

    // A set of attributes on the event.
    repeated AttributeKeyValue attributes = 3;
  }
```
Compared to OpenCensus this is 6x faster to encode, 3x faster to decode in Go in a simple benchmark with 3 events per span.

Reply:
I updated TimedEvent definition in the doc and added a benchmark. This is a nice improvement, thanks for the hint.